### PR TITLE
feat: Permissioned domains

### LINKED ISSUE #1833: Support Permissioned Domains (XLS-80d)
https://github.com/XRPLF/rippled/pull/5161

### PR DESCRIPTION
**Pull request by kuznetsss**
Fixes #1833.

### CODE REVIEW

**Review by PeterChen13579 — CHANGES_REQUESTED** [review]
Looks good 💯 🚀 . 
I noticed one edge case that is missing a test: both account and seq being invalid.
For this input 
```
{"method": "ledger_entry", "params": [ {
"permissioned_domain": {
"account": "",
"seq": -1
}
} ]}
```
Rippled returns :         "error": "malformedSequence",
while Clio returns:        "error": "malformedAccount",

File: src/rpc/Errors.cpp [hunk 1]
```diff
@@ -89,6 +89,11 @@ getErrorInfo(ClioError code)
         {.code = ClioError::RpcMalformedAuthorizedCredentials,
          .error = "malformedAuthorizedCredentials",
          .message = "Malformed authorized credentials."},
+        {.code = ClioError::RpcMalformedObject, .error = "malformedObject", .message = "Malformed object."},
+        {.code = ClioError::RpcMalformedObjectId, .error = "malformedObjectId", .message = "Malformed object ID."},
+        {.code = ClioError::RpcMalformedAccount, .error = "malformedAccount", .message = "Malformed account."},
```
Comment (cindyyan317) [inline src/rpc/Errors.cpp]: Maybe we can add //TODO here: replace when rippled side adds the error code. I think these errors should be added at rippled side.

File: tests/common/util/TestObject.cpp [hunk 2]
```diff
@@ -1166,6 +1166,30 @@ createMpTokenObject(std::string_view accountId, ripple::uint192 issuanceID, std:
     return mptoken;
 }
 
+ripple::STObject
+createPermissionedDomainObject(
+    std::string_view accountId,
+    ripple::LedgerIndex ledgerIndex,
```
Comment (cindyyan317) [inline tests/common/util/TestObject.cpp]: ripple::LedgerIndex is a uint32_t which represents ledger sequence. Here sfLedgerIndex expects a uint256, we usually pass it via string_view. Ledger sequence can not be used as index of ledger entry. I know the name is confusing 😵

**Comment by kuznetsss** [comment]
The PR https://github.com/XRPLF/rippled/pull/5252 in rippled was merged, so I will remove new errors here.

File: src/rpc/Errors.cpp (hunk 1 above)
Comment (kuznetsss) [reply src/rpc/Errors.cpp]: The PR https://github.com/XRPLF/rippled/pull/5252 in rippled was merged, so I will remove new errors here.

**Review by cindyyan317 — DISMISSED** [review]
LGTM

File: conanfile.py [hunk 3]
```diff
@@ -28,7 +28,7 @@ class Clio(ConanFile):
         'protobuf/3.21.9',
         'grpc/1.50.1',
         'openssl/1.1.1u',
-        'xrpl/2.4.0-b1',
+        'xrpl/2.4.0-b1@clio/permission_domains',
```
Comment (kuznetsss) [inline conanfile.py]: We should wait for `xrpl/2.4.0-b2` to merge this PR.

**Review by PeterChen13579 — DISMISSED** [review]
LGTM. 👍  Glad for the error msg changes.

**Review by cindyyan317 — APPROVED** [review]
